kernel/src/engine/arrow_utils.rs

```diff
@@ -459,6 +459,52 @@ fn get_indices(
     ))
 }
 
+/// Contains information about a StructField
```

Comment:
is there a reason to not just make this a struct? I too have gone down the "tuple all the things!" path, and generally have found it's cleaner to just make a struct

Reply:
It was a tuple before and I was trying to avoid doing a refactor + feature since this one is getting a bit long. I can switch it over if you think it's fine 👍

Reply:
Made two new structs. PTAL